### ISSUE TITLE
Crash when retrieving advertisingIdInfo

### ISSUE THREAD

**Issue by kazimierasc** [Iterable/iterable-android-sdk#67]
The `IterableApi` crashes on line 1281 on version 3.0.4 to some users.

The stacktrace:

```
java.lang.RuntimeException: 
  at android.app.ApplicationPackageManager.getPackageInfoAsUser (ApplicationPackageManager.java:178)
  at android.app.ApplicationPackageManager.getPackageInfo (ApplicationPackageManager.java:151)
  at com.google.android.gms.ads.identifier.AdvertisingIdClient.zza (Unknown Source:7)
  at com.google.android.gms.ads.identifier.AdvertisingIdClient.zza (Unknown Source:17)
  at com.google.android.gms.ads.identifier.AdvertisingIdClient.getAdvertisingIdInfo (Unknown Source:47)
  at com.iterable.iterableapi.IterableApi.getAdvertisingId (IterableApi.java:1281)
  at com.iterable.iterableapi.IterableApi.registerDeviceToken (IterableApi.java:1167)
  at com.iterable.iterableapi.IterableApi$2.run (IterableApi.java:530)
  at java.lang.Thread.run (Thread.java:784)
```

I've tried to look at the docs for the [AdvertisingIdClient](https://developers.google.com/android/reference/com/google/android/gms/ads/identifier/AdvertisingIdClient.html#getAdvertisingIdInfo(android.content.Context)). They don't document anything of the nature, so it may be some problem with them. Nevertheless, I suppose that some fallback would be reasonable here.

**Comment by vbabenkoru** [comment]
Thanks for the report! We'll add a more generic exception handler to catch those random errors.

**Comment by vbabenkoru** [comment]
Fixed, it'll be in the next update – 3.0.5
